app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,59 @@
+class RentalsController < ApplicationController
+  def checkout
+    # create new rental
+    rental = Rental.new(rental_params)
+
+    customer = Customer.find_by(id: params[:customer_id])
+    video = Video.find_by(id: params[:video_id])
+    
+    if !customer
+      render json: {
+          errors: ['Not Found']
+        }, status: :not_found
+        return
+    end
+      
+    if !video
+      render json: {
+          errors: ['Not Found']
+        }, status: :not_found
+        return
+    end
+        
+    if video.available_inventory == 0
+      render json: {
+          errors: ['Bad Request']
+        }, status: :bad_request
+      return
+    end
+
+    Rental.check_out(rental.customer_id, rental.video_id)
```

Comment:
I'm not sure if this would work, since rental hasn't been created yet, but could be an instance method instead of a class method? Then we wouldn't need to pass in parameters because the method would know that customer_id refers to rental.customer_id, etc.

```suggestion
    rental.check_out
```